src/search_stream.rs

```diff
@@ -203,6 +213,9 @@ impl<'a, R: io::Read, W: Terminal + Send> Searcher<'a, R, W> {
                 break;
             }
             while self.inp.pos < self.inp.lastnl {
+                if self.opts.files_with_matches && self.match_count > 0 {
+                    break;
```

Comment:
@BurntSushi in my experimentation, this seemed to work correctly to terminate after one match. However, I expected to be able to just break at the first match further down in the loop... but when I tried that I found that the match_count was > 1 at the top of the loop, which was unexpected. I'm missing something.